.travis.apple-third-tier.sh

```diff
@@ -96,7 +96,9 @@ then
      if [ "$(uname -m)" = "arm64" ]; then
          title "••••• Darwin: visionOS simulator tests •••••"
          title "boot a simulator"
-         xcrun simctl list devicetypes vision
+         xcrun simctl list devices vision
+         system_profiler SPDeveloperToolsDataType
+         xcodebuild -showsdks
```

Comment:
After looking into this a bit, the [current default xcode used on the `macOS-14` runner is 15.0.1](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md). The [visionOS simulator requires xcode 15.2+](https://developer.apple.com/documentation/xcode-release-notes/xcode-15_2-release-notes) and I don't feel like adding in the logic to switch between xcode versions in this CI.

Also, https://github.com/actions/runner-images/issues/9511

Reply:
Well, this still applies. I can either remove visionOS from the CI matrix or we can wait until the macos runner updates the default xcode. I'm not sure what the xcode versioning policy is (if there is one) on github runners.

Reply:
How about we do all the steps, including creating the simulator and deleting it, but making the steps that fail (presumably just lines 106 or 107) conditional:
```
if [ -z "${GITHUB_ACTIONS}" ]; then
   # These steps fail in GitHub Actions due to https://...
fi
```